### ISSUE TITLE
Jenkins framework crash if inside a docker container in bridge mode

### ISSUE THREAD

**Issue by gdippolito** [jenkinsci/mesos-plugin#236]
I'm experience this issue with the following setup:
Jenkins 2.2 (running in docker with bridge network)
Mesos-plugin 0.12
Mesos 0.28.1

Apparently the framework keep crashing after starting the connection with the mesos master. The framework is using the IP address of the container which is not reachable from the Mesos master (mesos master is running in a container in a different host). 

These are the logs on the mesos master:

```
I0510 10:32:34.468268    12 master.cpp:2231] Received SUBSCRIBE call for framework 'jenkins-framework' at scheduler-1d3da1d1-aa5d-4c78-a65a-ccb1dbfd5c84@172.17.0.3:45128
I0510 10:32:34.475852    12 master.cpp:2302] Subscribing framework jenkins-framework with checkpointing enabled and capabilities [  ]
I0510 10:32:34.483469     9 hierarchical.cpp:265] Added framework 0ccc0b5b-b484-4821-9ace-5eddda056dcf-0157
I0510 10:32:34.514787    12 master.cpp:5324] Sending 2 offers to framework 0ccc0b5b-b484-4821-9ace-5eddda056dcf-0157 (jenkins-framework) at scheduler-1d3da1d1-aa5d-4c78-a65a-ccb1dbfd5c84@172.17.0.3:45128
I0510 10:32:34.692304     9 master.cpp:2231] Received SUBSCRIBE call for framework 'jenkins-framework' at scheduler-1d3da1d1-aa5d-4c78-a65a-ccb1dbfd5c84@172.17.0.3:45128
I0510 10:32:34.692454     9 master.cpp:2302] Subscribing framework jenkins-framework with checkpointing enabled and capabilities [  ]
I0510 10:32:34.692483     9 master.cpp:2312] Framework 0ccc0b5b-b484-4821-9ace-5eddda056dcf-0157 (jenkins-framework) at scheduler-1d3da1d1-aa5d-4c78-a65a-ccb1dbfd5c84@172.17.0.3:45128 already subscribed, resending acknowledgement
I0510 10:32:35.919252     9 master.cpp:2231] Received SUBSCRIBE call for framework 'jenkins-framework' at scheduler-1d3da1d1-aa5d-4c78-a65a-ccb1dbfd5c84@172.17.0.3:45128
I0510 10:32:35.919471     9 master.cpp:2302] Subscribing framework jenkins-framework with checkpointing enabled and capabilities [  ]
I0510 10:32:35.919534     9 master.cpp:2312] Framework 0ccc0b5b-b484-4821-9ace-5eddda056dcf-0157 (jenkins-framework) at scheduler-1d3da1d1-aa5d-4c78-a65a-ccb1dbfd5c84@172.17.0.3:45128 already subscribed, resending acknowledgement
E0510 10:32:37.481415    13 process.cpp:1958] Failed to shutdown socket with fd 24: Transport endpoint is not connected
```

I have tried to change the Jenkins URL to the IP address of the VM where docker is running but it seems to make no changes. 

If I run the container with networking "host" mode I can successfully register the jenkins framework to the Mesos master.

**Comment by vinodkone** [comment]
I don't see the "crash" in the logs you posted?

Anyway a mesos frameowrk should advertise a routable IP to the mesos master, otherwise things don't work. You can set LIBPROCESS_ADVERTISE_IP and LIBPROCESS_ADVERTISE_PORT environment variables before starting jenkins master. These values should correspond to the the routable IP address and port, which in this case is the docker host's IP and host port forwarded to the container's port.

**Comment by geoffroyjabouley** [comment]
How are you starting your jenkins master? Are you using Marathon?

In bridged mode, your docker container shall know its host IP and run with at least **3 binded ports**:
1. PORT0 = jenkins http port
2. PORT1 = jenkins jnlp port
3. PORT2 = mesos libprocess port

Then you can set, when starting the container:
```
export LIBPROCESS_PORT="${PORT2}"
export LIBPROCESS_ADVERTISE_IP="${HOST_IP}"
export LIBPROCESS_ADVERTISE_PORT="${LIBPROCESS_PORT}"
```

**Comment by gdippolito** [comment]
By crash I means that the framework attempt to register to the Mesos master and immediately fail. 

There is a configuration option in the mesos-plugin section under "Manage Jenkins" called **Jenkins URL** I thought this option would have set up the correct Jenkins IP address and port to pass to Mesos master. I will try to pass this variable and I will let you know how it goes

**Comment by iax7** [comment]
I made a comment In another Issue but this seems to be the exact match to my problem, I paste it here also:
---
I'm running Mesos 0.28.2 and Jenkins 1.651.3 (also testing 2.7) with plugin version 0.13.0, (as I saw that 0.13.1 is tied to Mesos 1.0)

I'm getting disconnected. 
![image](https://cloud.githubusercontent.com/assets/6983510/18013816/890e9e3c-6b85-11e6-9faf-7c8cb128c6af.png)

**my Jenkins configuration is this:**
![image](https://cloud.githubusercontent.com/assets/6983510/18013875/d29b2534-6b85-11e6-8644-db398b07c626.png)

Not sure if its an incompatibility with the Mesos version or I have a wrong setting.

(My jenkins are containers, with mesos lib installed manually in each one)
![image](https://cloud.githubusercontent.com/assets/6983510/18014057/d0d2669e-6b86-11e6-987c-e47133b793f4.png)

**Comment by lavcraft** [comment]
@iax7 you have to use zookeeper url in `Mesos master [hostname:port]` field

In my example:
![screen shot 2017-01-10 at 19 49 24](https://cloud.githubusercontent.com/assets/576422/21947704/8e6d67b0-d9f7-11e6-9f34-b4c88e7475b5.png)

**Comment by kopax** [comment]
Hi, I have the same issue with a jenkins host running in bridged mode.

![image](https://cloud.githubusercontent.com/assets/1866564/25831747/5c76b1c8-3490-11e7-9ed4-c55ae371c89b.png)

It was working fine when I was trying in standalone mode. 

@geoffroyjabouley should I use the export you just wrote? What about `LIBPROCESS_IP` [here](http://stackoverflow.com/questions/27925909/jenkins-slave-in-mesos-not-start)

**Comment by jeschkies** [comment]
@gdippolito, were you able to resolve the issue?

**Comment by jeschkies** [comment]
Closing for lack of response.
